pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -56,42 +53,37 @@ func (l *latencyProfileObserver) observeLatencyProfile(
 	existingConfig map[string]interface{},
 ) (ret map[string]interface{}, errs []error) {
 
-	// traverse list of latencyConfigs to get all the config paths
-	configPaths := make([][]string, len(l.latencyConfigs))
-	for i, latencyConfig := range l.latencyConfigs {
-		configPaths[i] = latencyConfig.ConfigPath
-	}
-
 	defer func() {
 		// Prune the observed config so that it only contains fields specific to this observer
-		ret = configobserver.Pruned(ret, configPaths...)
+		ret = configobserver.Pruned(ret, l.usedConfigPaths...)
 	}()
 
+	for _, shouldSupressConfigUpdatesFn := range l.shouldSuppressConfigUpdatesFuncs {
+		suppress, reason, err := shouldSupressConfigUpdatesFn()
+		if err != nil {
+			errs = append(errs, err)
+			return existingConfig, errs
+		}
+		if suppress {
+			// log that latency profile couldn't be updated due to conditional
+			klog.Warningf("latency profile config observer suppressed update to observed config: %s", reason)
```

Comment:
can't decide if this logging is too much or not. For example `not all static pod revisions match` is a normal flow and could clutter the log with too many messages.

I think we should at least tune it down a bit, to info level. At least the information will be there if somebody will be looking for it, but will not be that aggressive.

Reply:
Hmm, updated to use `klog.Infof` instead of `klog.Warningf`.